gc/gc-base/src/main/java/org/projectnessie/gc/base/DistributedIdentifyContents.java

```diff
@@ -66,21 +69,26 @@ public DistributedIdentifyContents(SparkSession session, GCParams gcParams) {
    *
    * @param liveContentsBloomFilterMap live contents bloom filter per content id.
    * @param references list of all the references to walk (live and dead)
-   * @return {@link IdentifiedResult} object.
+   * @return current run id of the completed gc task
    */
-  public IdentifiedResult getIdentifiedResults(
+  public String identifyExpiredContents(
       Map<String, ContentBloomFilter> liveContentsBloomFilterMap, List<Reference> references) {
 
     IdentifyContentsPerExecutor executor = new IdentifyContentsPerExecutor(gcParams);
-    List<IdentifiedResult> results =
+    JavaRDD<IdentifiedResult> resultRDD =
```

Comment:
why are we using a RDD instead of a dataframe. This low level API should be avoided

Reply:
When I tried using DataFrame instead of RDD, DataFrame is doing codeGen with Nessie's `Reference` interface. As `Reference` doesn't have public constructor, code gen failed. If I add a public constructor with immutables, Immutables is failing to build the class saying it cannot apply validation. Need to explore further. Instead of helping Dataframe is causing problem.